### PR TITLE
[Feature Highlight] Swipe to dismiss

### PR DESCRIPTION
**Pull request by samnm**

### CODE REVIEW

File: components/FeatureHighlight/src/private/MDCFeatureHighlightView+Private.m [hunk 1]
```diff
@@ -17,10 +17,18 @@
...
+#import "MDCFeatureHighlightDismissGestureRecognizer.h"
+#import "MDFTextAccessibility.h"
+
 #import "MaterialFeatureHighlightStrings.h"
 #import "MaterialFeatureHighlightStrings_table.h"
+#import "MaterialMath.h"
 #import "MaterialTypography.h"
 
+static inline CGFloat CGPointDistanceToPoint(CGPoint a, CGPoint b) {
+  return sqrt(pow(a.x - b.x, 2) + pow(a.y - b.y, 2));
```
Comment (ianegordon) [inline components/FeatureHighlight/src/private/MDCFeatureHighlightView+Private.m]: MDCHypot(a.x - b.x, a.y - b.y)

File: components/FeatureHighlight/src/private/MDCFeatureHighlightDismissGestureRecognizer.m [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+- (void)reset {
+  [super reset];
+
+  _hasTouch = NO;
+  _velocity = 0;
+}
+
+- (void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event {
+  assert([self.view isKindOfClass:[MDCFeatureHighlightView class]]);
```
Comment (ianegordon) [inline components/FeatureHighlight/src/private/MDCFeatureHighlightDismissGestureRecognizer.m]: s/assert/NSCAssert/

**Review by ianegordon — CHANGES_REQUESTED** [review]

File: components/FeatureHighlight/src/private/MDCFeatureHighlightView+Private.m (hunk 1 above)
Comment (samnm) [reply components/FeatureHighlight/src/private/MDCFeatureHighlightView+Private.m]: Done.

File: components/FeatureHighlight/src/private/MDCFeatureHighlightDismissGestureRecognizer.m (hunk 2 above)
Comment (samnm) [reply components/FeatureHighlight/src/private/MDCFeatureHighlightDismissGestureRecognizer.m]: Done.

**Comment by ianegordon** [comment]
The Travis failures are caused by floating point conversion errors.
Please investigate.

**Comment by ianegordon** [comment]
Verified VO Back gesture works as expected on iPhone 5 w/ iOS 8.3.

**Review by ianegordon — APPROVED** [review]

**Comment by ianegordon** [comment]
Tested Locally.

**Pull request merged**